nix/find-haskell-paths/parser.nix

```diff
@@ -44,4 +44,44 @@ in
         val;
     in
     parsec.runParser parser packageYamlFile;
+
+  # Extract all the executables from a .cabal file 
+  parseCabalExecutableName = cabalFile:
+    with parsec;
+    let
+      parseExec =
+        bind anyChar (a:
+          bind anyChar (b:
+            bind anyChar (c:
+              bind anyChar (d:
+                bind anyChar (e:
+                  bind anyChar (f:
+                    bind anyChar (g:
+                      bind anyChar (h:
+                        bind anyChar (i:
+                          bind anyChar (j:
+                            if a == "e" &&
+                              b == "x" &&
+                              c == "e" &&
+                              d == "c" &&
+                              e == "u" &&
+                              f == "t" &&
+                              g == "a" &&
+                              h == "b" &&
+                              i == "l" &&
+                              j == "e"
+                            then fail else (pure null)
+                          ))))))))));
```

Comment:
What is the purpose of this verbose `parseExec`? Why not use `parsec.string "executable"`?

Reply:
You could probably obviate thus using [`skipTill1`](https://github.com/kanwren/nix-parsec/blob/1bf25dd9c5de1257a1c67de3c81c96d05e8beb5e/parsec.nix#L566). @ursi might have better ideas.

Reply:
I know of know reason to use this over `parsec.string`. Regarding anything fancier than than, I'd have to brush up on the library. In general my takeaway from the purescript parser was to use `parsec.regex` whenever possible for anything more complicatex than exact matches. If was much more performant.

Reply:
Have pushed the change that uses `parsec.string`